pkg/controller/replication_controller.go

```diff
@@ -198,20 +199,43 @@ func FilterActivePods(pods []api.Pod) []api.Pod {
 	return result
 }
 
+type activePods []api.Pod
+
+func (s activePods) Len() int      { return len(s) }
+func (s activePods) Swap(i, j int) { s[i], s[j] = s[j], s[i] }
+
+func (s activePods) Less(i, j int) bool {
+	// Not ready < ready
+	if !api.IsPodReady(&s[i]) && api.IsPodReady(&s[j]) {
```

Comment:
I would order the checks as follows:
1. host
2. phase
3. readiness

but it shouldn't make a lot of difference.